test/bin/depext.t/run.t

```diff
@@ -22,7 +22,11 @@ We lock and expect the OS package to be part of the locked Opam file.
   libfantasydependency
 
 Then we want to make sure `depext` works. Given the fantasy package does not
-exist, `depext` should not install any packages but still work successfully:
+exist, `depext` should not install any packages but still work successfully.
+On some systems, opam reports non-existent packages as available, and such
+packages will appear in the output of `opam-monorepo depext --dry-run`. Filter
+lines containing "libfantasydependency" from the output for consistency across
+systems.
 
-  $ opam-monorepo depext --dry-run
+  $ opam-monorepo depext --dry-run | grep -v libfantasydependency
```

Comment:
I've seen this test fail but I fear there's something else going on because it didn't use to fail on my system but since recently it does. It seems the behaviour of OPAM changed at some point.

If my backtracking is correct, this changed between OPAM 2.1.0 and 2.1.1, so we should probably make `opam-monorepo depext` work consistently either on all supported versions of OPAM or raise the dependency and add the behaviour change in a PR (with a changelog entry since it changes user-visible behavior).

Reply:
OPAM separates depexts into available, unavailable, and installed. For some package managers, all non-installed packages are assumed to be available. Homebrew is one such package manager. It looks like we should be able to quickly query the brew package repo for available packages using a regexp, similar to how we do it for apt. I'll make a PR to opam to add this.

Reply:
Here's the PR that lets opam check for available packages with homebrew: https://github.com/ocaml/opam/pull/5372

Reply:
Is that separation something that has been added in 2.1.1? Because on my Fedora Linux at hand the behavior of `opam-monorepo` changes whether using opam-core 2.1.0 or 2.1.1, so it seems like the availability check was removed in that case which I think is would be odd.

Reply:
I did some archeology and it looks like support for checking for available packages on fedora and friends was remove intentionally between 2.1.0 and 2.1.1 in this commit https://github.com/ocaml/opam/commit/c76a512a4d887a6e9623c6f5447ff8d26104cc49. It's unrelated to the lack of support for checking for available packages on macos with homebrew.

Reply:
Ah, ok. I see. Thanks for the investigation. I feared this would be a general change in OPAM and while I am not super happy about this happening in a patch version I guess there's not much worth doing on our side.